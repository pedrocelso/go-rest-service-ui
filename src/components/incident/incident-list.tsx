import React, { Component } from 'react';
import { connect } from 'react-redux'
import { map, pipe, sort, reverse } from 'ramda'
import { Grid, Typography, Paper } from '@material-ui/core';

import { deauthenticate } from '../login/login-actions'
import { Incident } from '../../services/task';
import { formatTime, sortByDate } from '../../modules/date';

interface Props {
  deauthenticate: typeof deauthenticate
  incidentList: Incident[];
}

export class IncidentList extends Component<Props, {}> {
  render() {
    const sortIncidents = (l: Incident[]): Incident[] => sort((a, b) => b.status - a.status, l)
    const incidentList = pipe(
      // @ts-ignore until curry and pipe works fine with TS
      sortByDate(`creationTime`) as Incident[],
      reverse,
      sortIncidents
    )(this.props.incidentList)

    return (
      <div className="incident__paper">
        {map((t) => {
          const pendingPill = (<span className="badge badge-primary badge-pill">Pending</span>)
          const isPending = t.status === 1
          const nameEl = (
            <Typography gutterBottom variant="subtitle1" color={isPending ? `error` : `textPrimary`}>
              {t.name} {isPending ? pendingPill : null}
            </Typography>
          )

          return (
            <Paper elevation={2} square={true} key={t.id}>
              <Grid container className="card__grid">
                <Grid item xs={12} sm container>
                  <Grid item xs container direction="column">
                    <Grid item xs>
                      {nameEl}
                      <Typography className="d-none d-lg-block" gutterBottom>{t.description}</Typography>
                      <Typography color="textSecondary">Created: {formatTime(t.creationTime)}</Typography>
                      <Typography color="textSecondary">Updated: {formatTime(t.updateTime)}</Typography>
                    </Grid>
                  </Grid>
                </Grid>
              </Grid>
            </Paper>
          )
        }, incidentList)}
      </div>
    )
  }
}

const mapDispatchToProps = { deauthenticate }
export default connect(null, mapDispatchToProps)(IncidentList)