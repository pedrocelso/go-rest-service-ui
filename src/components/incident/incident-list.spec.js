import React from 'react';
import {shallow} from 'enzyme'
import sinon from 'sinon'

import {IncidentList} from './incident-list'

describe(`<IncidentList />`, () => {
  it(`Should render the list`, () => {
    const incidentList = [
      { id: 10, name: `test`, description: `test@test.com`, creationTime: 1556924666404, updateTime: 1556924566404, status: 0 },
      { id: 11, name: `test1`, description: `test1@test.com`, creationTime: 1556924666404, updateTime: 0, status: 1 },
    ]

    const component = shallow(
      <IncidentList incidentList={incidentList} />,
    );

    expect(component).toMatchSnapshot();
  });

  it(`Should render an empty list`, () => {
    const incidentList = []

    const component = shallow(
      <IncidentList incidentList={incidentList} />,
    );

    expect(component).toMatchSnapshot();
  });

});
