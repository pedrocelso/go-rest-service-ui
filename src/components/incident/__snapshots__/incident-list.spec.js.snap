// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IncidentList /> Should render an empty list 1`] = `
<div
  className="incident__paper"
/>
`;

exports[`<IncidentList /> Should render the list 1`] = `
<div
  className="incident__paper"
>
  <WithStyles(Paper)
    elevation={2}
    key="10"
    square={true}
  >
    <WithStyles(Grid)
      className="card__grid"
      container={true}
    >
      <WithStyles(Grid)
        container={true}
        item={true}
        sm={true}
        xs={12}
      >
        <WithStyles(Grid)
          container={true}
          direction="column"
          item={true}
          xs={true}
        >
          <WithStyles(Grid)
            item={true}
            xs={true}
          >
            <WithStyles(Typography)
              color="default"
              gutterBottom={true}
              variant="subtitle1"
            >
              test
               
            </WithStyles(Typography)>
            <WithStyles(Typography)
              className="d-none d-lg-block"
              gutterBottom={true}
            >
              test@test.com
            </WithStyles(Typography)>
            <WithStyles(Typography)
              color="textSecondary"
            >
              Created: 
              2019/05/03, 4:04:26 pm
            </WithStyles(Typography)>
            <WithStyles(Typography)
              color="textSecondary"
            >
              Updated: 
              2019/05/03, 4:02:46 pm
            </WithStyles(Typography)>
          </WithStyles(Grid)>
        </WithStyles(Grid)>
      </WithStyles(Grid)>
    </WithStyles(Grid)>
  </WithStyles(Paper)>
  <WithStyles(Paper)
    elevation={2}
    key="11"
    square={true}
  >
    <WithStyles(Grid)
      className="card__grid"
      container={true}
    >
      <WithStyles(Grid)
        container={true}
        item={true}
        sm={true}
        xs={12}
      >
        <WithStyles(Grid)
          container={true}
          direction="column"
          item={true}
          xs={true}
        >
          <WithStyles(Grid)
            item={true}
            xs={true}
          >
            <WithStyles(Typography)
              color="error"
              gutterBottom={true}
              variant="subtitle1"
            >
              test1
               
              <span
                className="badge badge-primary badge-pill"
              >
                Pending
              </span>
            </WithStyles(Typography)>
            <WithStyles(Typography)
              className="d-none d-lg-block"
              gutterBottom={true}
            >
              test1@test.com
            </WithStyles(Typography)>
            <WithStyles(Typography)
              color="textSecondary"
            >
              Created: 
              2019/05/03, 4:04:26 pm
            </WithStyles(Typography)>
            <WithStyles(Typography)
              color="textSecondary"
            >
              Updated: 
              n/a
            </WithStyles(Typography)>
          </WithStyles(Grid)>
        </WithStyles(Grid)>
      </WithStyles(Grid)>
    </WithStyles(Grid)>
  </WithStyles(Paper)>
</div>
`;
